go/api/handlers/document.go

```diff
@@ -0,0 +1,189 @@
+	"challenge-flutter-go/api/utils"
+	"challenge-flutter-go/models"
+	"challenge-flutter-go/repository"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/gin-gonic/gin"
+)
+
+type DocumentHandler struct {
+	Repository     repository.DocumentRepository
+	TripRepository repository.TripRepository
+}
+
+// @Summary		Get all documents of a trip
+// @Description	Get all documents of a trip
+// @Tags			document
+// @Accept			json
+// @Produce		json
+// @Security		BearerAuth
+// @Param			id		path		string	true	"ID of the trip"
+// @Failure		400		{object}	error
+// @Failure		401		{object}	error
+// @Router			/trips/{id}/documents [get]
+func (handler *DocumentHandler) GetDocumentsOfTrip(context *gin.Context) {
+	tripId := context.Param("id")
+
+	trip, _ := handler.TripRepository.Get(tripId)
+
+	documents, err := handler.Repository.GetDocuments(trip)
+	if err != nil {
+		errorHandlers.HandleGormErrors(err, context)
+		return
+	}
+
+	documentResponses := make([]responses.DocumentResponse, len(documents))
+	for i, document := range documents {
+		documentResponses[i] = responses.DocumentResponse{
+			ID:          document.ID,
+			Title:       document.Title,
+			Description: document.Description,
+			Path:        document.Path,
+			CreatedAt:   document.CreatedAt.Format(time.RFC3339),
+			UpdateAt:    document.UpdatedAt.Format(time.RFC3339),
+		}
+	}
+	context.JSON(http.StatusOK, documentResponses)
+}
+
+// @Summary		Create a new document on trip
+// @Description	Create a new document on trip
+// @Tags			document
+// @Accept			multipart/form-data
+// @Produce		json
+// @Security		BearerAuth
+// @Param			id		path		string	true	"ID of the trip"
+// @Param			title		formData	string	true	"Title of the document"
+// @Param			description	formData	string	true	"Description of the document"
+// @Param			document	formData	file	true	"Document file"
+// @Success		200		{object}	responses.DocumentResponse
+// @Failure		400		{object}	error
+// @Failure		401		{object}	error
+// @Router			/trips/{id}/documents [post]
+func (handler *DocumentHandler) CreateDocument(context *gin.Context) {
+	context.Request.Body = http.MaxBytesReader(context.Writer, context.Request.Body, 10<<20)
+
+	tripId := context.Param("id")
+
+	title := context.PostForm("title")
+	description := context.PostForm("description")
+
+	file, errFile := context.FormFile("document")
+	if errFile != nil {
+		context.JSON(http.StatusBadRequest, gin.H{"error": "Document is required"})
+	}
+
+	if file.Size > 5<<20 {
+		context.JSON(http.StatusBadRequest, gin.H{"error": "File size exceeds limit of 5 MB"})
+		return
+	}
+
+	filePath, errFilePath := utils.SaveUploadedFile(file, "document", title)
+	if errFilePath != nil {
+		fmt.Println(errFilePath)
+		context.JSON(http.StatusBadRequest, gin.H{"error": "Error saving file"})
+		return
+	}
+
+	trip, _ := handler.TripRepository.Get(tripId)
+	var document = models.Document{
+		Title:       title,
+		Description: description,
+		TripID:      trip.ID,
+		Path:        filePath,
+	}
```

Comment:
```suggestion
	var document = models.Document{
		Title:       title,
		Description: description,
		TripID:      tripId,
		Path:        filePath,
	}
```

Une query sur la database en moins